R/utilities.R

```diff
@@ -2236,4 +2236,175 @@ SparseSpearmanCor <- function(X, Y = NULL, cov = FALSE) {
     }
   rankY <- SparsifiedRanks(X = Y)
   return(qlcMatrix::corSparse(X = rankX, Y = rankY, cov = cov))
-}
\ No newline at end of file
+}
+
+# Export bigwig file for each group of bed file present in outdir
+# @param object The seurat object
+# @param assay The assay of the fragment file
+# @param group.by The metadata used to split the fragment file
+# @param idents The idents from the group.by to be exported
+# @param normMethod Normalization method for the biwig files
+# It can be the name of any quantitative column in the @meta.data
+# or ncells as the number of cells
+# @param tileSize The size of the tiles in the bigwig file
+# @param minCells The minimum of cells in a group to be exported
+# @param cutoff The maximum number of fragment in a given tile
+# @param chromosome The chromosomes to be exported 
+# @param threads Number of threads
+# @param outdir The output directory for bigwig file
+# Also used as output directory for SlitFragment function
+# @param verbose Display messages
+#' @importFrom GenomicRanges seqlengths GRanges slidingWindows
+#' @importFrom future plan
+#' @importFrom future.apply future_lapply
+
+ExportGroupBW  <- function(
+  object = NULL,
+  assay = "peaks",
+  group.by = "seurat_clusters",
+  idents = c(0,2),
+  normMethod = "TSS.enrichment",
```

Comment:
I think we should just have a depth normalization by default, and I'd be ok with having additional options if you want to include TSS norm or option to use any numeric value in the metadata. Keep in mind that people can use this software for things other than ATAC, so TSS enrichment is not always expected. I think the default should be fragments per million (or 100k), with the total fragments coming from the size of the fragment file rather than `nCount_ATAC` which is based on the count matrix

Reply:
Let me know if this fit what you had in mind for the normalization